backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -118,6 +128,14 @@ public List<VdsCpuUnit> updatePhysicalCpuAllocations(VM vm, Map<Guid, List<VdsCp
 
         filterSocketsWithInsufficientMemoryForNumaNode(cpuTopology, vm, hostId);
         List<VdsCpuUnit> cpusToBeAllocated = allocateDedicatedCpus(cpuTopology, vm, hostId);
+        if (vm.getCpuPinningPolicy() == CpuPinningPolicy.ISOLATE_THREADS) {
+            List<Integer> socketsUsed = cpusToBeAllocated.stream().map(VdsCpuUnit::getSocket).distinct().collect(Collectors.toList());
+            int cpus = 0;
+            for (int socket: socketsUsed) {
```

Comment:
Is cpus actually number of hostPhysicalCores used?

Why are you iterating over sockets and then getting distinct cores from socket - cpusToBeAllocated.stream().map(VdsCpuUnit::getCore).distinct().count() would not work?

Reply:
I will probably change it once I will work on Arik's incoming comments :)